libnetwork/driver_plugin.py

```diff
@@ -446,22 +425,26 @@ def join():
         # configured on the endpoint (which will be our host IPs).
         app.logger.debug("Using Calico IPAM driver, configure gateway and "
                          "static routes to the host")
-        ep = client.get_endpoint(hostname=hostname,
-                                 orchestrator_id=ORCHESTRATOR_ID,
-                                 workload_id=CONTAINER_NAME,
-                                 endpoint_id=endpoint_id)
         static_routes = []
-        if ep.ipv4_gateway:
-            json_response["Gateway"] = str(ep.ipv4_gateway)
+        if gateway_ip4:
+            json_response["Gateway"] = "169.254.1.1"
             static_routes.append({
-                "Destination": str(IPNetwork(ep.ipv4_gateway)),
+                "Destination": "169.254.1.1/32",
                 "RouteType": 1,  # 1 = CONNECTED
                 "NextHop": ""
             })
-        if ep.ipv6_gateway:
-            json_response["GatewayIPv6"] = str(ep.ipv6_gateway)
+        if gateway_ip6:
+            # Find which link local was assigned to the ipv6 interface
+            try:
+                ip_addr_output = check_output(["ip", "-6", "addr", "show", "dev", veth_name_host])
+                next_hop_6 = re.search(IPV6_RE, ip_addr_output).group(1)
```

Comment:
Once you get this working, might be worth communizing some of this code in libcalico.